lib/internal/modules/esm/worker.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const {
+  Int32Array,
+  ReflectApply,
+  SafeWeakMap,
+  TypedArrayPrototypeFill,
+  TypedArrayPrototypeSet,
+  Uint8Array,
+  globalThis: {
+    Atomics,
+  },
+} = primordials;
+const {
+  ERR_INVALID_RETURN_VALUE,
+} = require('internal/errors').codes;
+
+// Create this WeakMap in js-land because V8 has no C++ API for WeakMap.
+internalBinding('module_wrap').callbackMap = new SafeWeakMap();
+
+const { workerData } = require('worker_threads');
+// lock = 0 -> main sleeps
+// lock = 1 -> worker sleeps
+const lock = new Int32Array(workerData.lock); // Required by Atomics
+const data = new Uint8Array(workerData.data); // For v8.deserialize/serialize
+const CHUNK_THRESHOLD = data.length - 26; // `{ done: true/false, value: … }` adds 26 bytes
+
+function releaseLock() {
+  Atomics.store(lock, 0, 1); // Send response to main
```

Comment:
```suggestion
function releaseLock(data = 1) {
  Atomics.store(lock, 0, data); // Send response to main
```

Reply:
I don’t understand this one. Since we always use `1`, why would we accept input?

Reply:
I was using the lock to pass the length of the data to transfer (we don't have to always use `1`, we could use any non-zero 32-bit value to unlock the other thread)

Reply:
Okay so with the `postMessage` approach, we wouldn’t make this change? Since we don’t need to know the length.